packages/toolkit/src/tests/createSlice.test.ts

```diff
@@ -678,7 +678,7 @@ describe('createSlice', () => {
         initialState: [] as any[],
         reducers: (create) => ({
           thunkReducers: create.asyncThunk(
-            function payloadCreator(arg, api) {
+            function payloadCreator(arg: string, api) {
```

Comment:
Hmmm... This brings up the question: will this still be okay in an untyped `.js` file for non-TS-users or will it make it unusable?

Reply:
it's the same behaviour as `createAsyncThunk`. if they're not using type checking on their JS there will be nothing to complain, and if they are then they'll need to annotate:
```js
const getPost = createAsyncThunk(
  'posts/getPost',
  /**
   *
   * @param {number} id
   * @returns {Promise<Post>}
   */
  async (id) => {
    const response = await fetch(
      `https://jsonplaceholder.typicode.com/posts/${id}`,
    )
    const data = await response.json()
    return data
  },
)
```

Reply:
you can even set the ApiConfig:
```js
const getPost = createAsyncThunk(
  'posts/getPost',
  /** @type {import("@reduxjs/toolkit").AsyncThunkPayloadCreator<Post, number, { state: RootState }>}*/
  async (id, api) => {
    const response = await fetch(
      `https://jsonplaceholder.typicode.com/posts/${id}`,
    )
    const data = await response.json()
    return data
  },
)
```